### ISSUE TITLE
Reconnect doesn't work after an extended period

### ISSUE THREAD

**Issue by Jbonavita** [Benedicht/BestHTTP-Issues#66]
I'm using the following RetryPolicy but it doesn't reconnect, after say, 1 minute of the server being off.

`
public sealed class DefaultRetryPolicy : IRetryPolicy
    {
        private static TimeSpan?[] DefaultBackoffTimes = new TimeSpan?[]
        {
            TimeSpan.Zero,
            TimeSpan.FromSeconds(2),
            TimeSpan.FromSeconds(10),
            TimeSpan.FromSeconds(10),
            TimeSpan.FromSeconds(10)
        };

        TimeSpan?[] backoffTimes;

        public DefaultRetryPolicy()
        {
            this.backoffTimes = DefaultBackoffTimes;
        }

        public DefaultRetryPolicy(TimeSpan?[] customBackoffTimes)
        {
            this.backoffTimes = customBackoffTimes;
        }

        public TimeSpan? GetNextRetryDelay(RetryContext context)
        {
            if (context.PreviousRetryCount >= this.backoffTimes.Length)
                return TimeSpan.FromSeconds(10);

            return this.backoffTimes[context.PreviousRetryCount];
        }
    }`

**Comment by Benedicht** [comment]
Just tried it out, and it worked as expected!

As you can see in this charles log when the server shut down it started to try to connect to the server:
![image](https://user-images.githubusercontent.com/4248416/123466315-7ca4b080-d5ef-11eb-92a2-1b9bf8e8adba.png)

And after starting the server again, it's reconnected successfully:
![image](https://user-images.githubusercontent.com/4248416/123474255-f9d52300-d5f9-11eb-85d6-8907d43388c9.png)

Could you [set the plugin's logging level to All](https://besthttp-documentation.readthedocs.io/en/latest/#7.GlobalTopics/Logging/) and send over the produced log entries?

**Comment by Jbonavita** [comment]
I attached the log.

I turned off the server shortly after sending a test message and receiving it.
________________________________

**Comment by Jbonavita** [comment]
Use this log instead.

Here's the retry class:

    public sealed class DefaultRetryPolicy : IRetryPolicy
    {
        private static TimeSpan?[] DefaultBackoffTimes = new TimeSpan?[]
        {
            TimeSpan.Zero,
            TimeSpan.FromSeconds(2),
            TimeSpan.FromSeconds(10),
            TimeSpan.FromSeconds(10),
            TimeSpan.FromSeconds(10)
        };

        TimeSpan?[] backoffTimes;

        public DefaultRetryPolicy()
        {
            this.backoffTimes = DefaultBackoffTimes;
        }

        public DefaultRetryPolicy(TimeSpan?[] customBackoffTimes)
        {
            this.backoffTimes = customBackoffTimes;
        }

        public TimeSpan? GetNextRetryDelay(RetryContext context)
        {
            if (context.PreviousRetryCount >= this.backoffTimes.Length)
                return TimeSpan.FromSeconds(10);

            return this.backoffTimes[context.PreviousRetryCount];
        }
    }
________________________________

**Comment by Benedicht** [comment]
@Jbonavita i think the attachement is removed by github. Could you resend it to my email address (besthttp@gmail.com)?

**Comment by Jbonavita** [comment]
Emailed

**Comment by Benedicht** [comment]
The authenticator fails with _404 - Not Found_.

**Comment by Jbonavita** [comment]
Yes, because I turned the server off. I go into IIS and click Stop. I leave it stopped for a couple of minutes and then start the server again.

**Comment by Benedicht** [comment]
The log contains only the Error entries and it contains only the failed attempts. I would advise again to set the [log level to All](https://besthttp-documentation.readthedocs.io/en/latest/#7.GlobalTopics/Logging/), because that way the plugin going to log more details about what's happening.

**Comment by Jbonavita** [comment]
I emailed another log

**Comment by Benedicht** [comment]
It's reproducable only when there's an authenticator with `IsPreAuthRequired` set to true.
